### PR TITLE
BE for stacked area viz for #20004 (#18676)

### PR DESCRIPTION
**Pull request by howonlee**
More stuff in #18676. Supposed to be the BE for #20004 Just continuing static viz work. Stacked graphs only have a simple viz-settings flag turned on so making sure that gets to FE bit.

### CODE REVIEW

**Comment by howonlee** [comment]
What is actually gonna be tomorrow (in my sleep schedule, if not according to pacific time) is gonna be extensive testing to see if it's compatible with the FE

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

**Comment by alxnddr** [comment]
It did not stack for me for some reason, could you double check @howonlee?
Screenshot 2022-01-31 at 23 38 09
Screenshot 2022-01-31 at 23 38 16

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

File: src/metabase/pulse/render/body.clj [hunk 1]
```diff
@@ -348,12 +348,27 @@
...
+     :y        {:type (or (:graph.y_axis.scale viz-settings) "linear")
+                :format y-format }
+     :labels   labels}))
+
+(defn- set-default-stacked
+  "Default stack type is stacked for area chart with more than one metric.
+   So, if :stackable.stack_type is not specified, it's stacked.
+   However, if key is explicitly set in :stackable.stack_type and is nil, that indicates not stacked."
+  [viz-settings card]
+  (let [is-stacked     (if (contains? viz-settings :stackable.stack_type)
```
Comment (dpsutton) [inline src/metabase/pulse/render/body.clj:364]: `is-stacked` reads so strangely in a language that allows questions marks in identifiers. Minor nit but `stacked?` is far preferable

Comment (howonlee) [reply src/metabase/pulse/render/body.clj:364]: it feels like `stacked?` is an actual predicate function, tho, which this one is not. I will change it, tho

Comment (dpsutton) [reply src/metabase/pulse/render/body.clj:364]: Also, this logic here is careful to check that if there is a value at `(:stackable.stack_type viz-settings)` that it needs to be equal to `"stacked"`. But this subtlety is dropped for `default-y-pos` which only cares if there is _any_ truthy value at that location.

Comment (dpsutton) [reply src/metabase/pulse/render/body.clj:364]: > it feels like `stacked?` is an actual predicate function, tho, which this one is not. I will change it, tho

That's fair. I think some people make a big distinction about functions which return booleans vs boolean values themselves. I don't make a huge different. I think its a minor quibble so you can leave it alone if you like. But the `is-` prefix just stands out to me a lot. No big deal though.

Comment (howonlee) [reply src/metabase/pulse/render/body.clj:364]: possible values are `stacked`, `normalized` and nil. normalized is not implemented in FE yet and will require a fair bit of work in BE also.

Comment (howonlee) [reply src/metabase/pulse/render/body.clj:364]: actually, christ

`grep -nr '!= ' . | wc`
     331    2416   29835

**Review by dpsutton — APPROVED** [review]

**Review by alxnddr — APPROVED** [review]
Awesome 👏

**Comment by rlotun** [comment]
@metabase-bot backport release-x.42.x

**Pull request merged**